liquid-base/src/Foreign/ForeignPtr.spec

```diff
@@ -3,9 +3,10 @@ module spec Foreign.ForeignPtr where
 import GHC.ForeignPtr
 import Foreign.Ptr
 
-Foreign.ForeignPtr.withForeignPtr :: forall a b. fp:(GHC.ForeignPtr.ForeignPtr a) 
-  -> ((PtrN a (fplen fp)) -> GHC.Types.IO b) 
-  -> (GHC.Types.IO b)
+// `withForeignPtr` is missing from the `base` version shipped with GHC 9.0.1.
+// Foreign.ForeignPtr.withForeignPtr :: forall a b. fp:(GHC.ForeignPtr.ForeignPtr a)
+//   -> ((PtrN a (fplen fp)) -> GHC.Types.IO b)
+//   -> (GHC.Types.IO b)
```

Comment:
I believe that [the CI failures](https://app.circleci.com/pipelines/github/ucsd-progsys/liquidhaskell/1148/workflows/b46da860-8689-4faf-8233-a8ebcd957b01) for `stack_810` and `cabal_810` are caused by this change.

To do: investigate how this spec can be used for GHC 8.10 only.

Reply:
Given that `liquid-base` is building with both compilers in the `develop` branch, do you know if there is a change in this PR that started causing this error with ghc-9?

Reply:
Hm, I reverted the change (so `ForeignPtr.spec` is now as in `develop`), and CI passes -- as expected, since I I did not (intentionally) change any behaviour in this PR.

Reply:
This is puzzling. I created the following file, `stack-901.yaml`:

```yaml
flags:
  liquidhaskell:
    devel: true
  liquid-platform:
    devel: true
extra-package-dbs: []
ghc-options:
  hscolour: -w
allow-newer: true # 8.10.1
packages:
- liquid-fixpoint
- liquid-ghc-prim
- liquid-base
- liquid-bytestring
- liquid-prelude
- liquid-vector
- liquid-containers
- liquid-parallel
- liquid-platform
- .
extra-deps:
- hashable-1.3.0.0
- rest-rewrite-0.1.1

resolver: nightly-2021-11-28
compiler: ghc-9.0.1

nix:
  shell-file: shell-stack.nix
  path: ["nixpkgs=./nixpkgs.nix"]
```

When I build using this file, I get:

```
$ stack build --stack-yaml stack-901.yaml
[...]
liquid-base      >       
liquid-base      > **** LIQUID: SAFE (0 constraints checked) **************************************
liquid-base      > [159 of 214] Compiling Foreign.ForeignPtr
liquid-base      >       
liquid-base      > /.../liquidhaskell/liquid-base/src/Foreign/ForeignPtr.spec:6:1: error:
liquid-base      >     • Unknown variable `Foreign.ForeignPtr.withForeignPtr`
liquid-base      >     rawTySigs
liquid-base      >     • 
liquid-base      >   |   
liquid-base      > 6 | Foreign.ForeignPtr.withForeignPtr :: forall a b. fp:(GHC.ForeignPtr.ForeignPtr a) 
liquid-base      >   |
```

I now realise that there is no Stack + GHC 9 setup in CI, only Cabal + GHC 9. Perhaps the issue is with the Stack setup I created locally to test this ...

Reply:
When I build with Stack using the YAML above (sans the `liquidhaskell:devel` flag) on `develop`, it fails too, with the same error. I don't understand what's going on there, but this points to the issue being unrelated to this PR. So I think we should merge this PR.

Reply:
Awesome thanks for looking into this!